apps/server/src/modules/synchronization/uc/synchronization.uc.ts

```diff
@@ -0,0 +1,106 @@
+import { UserService } from '@modules/user';
+import { Injectable } from '@nestjs/common';
+import { Logger } from '@src/core/logger';
+import { SanisResponse, SchulconnexRestClient } from '@src/infra/schulconnex-client';
+import { ConfigService } from '@nestjs/config';
+import { SynchronizationService } from '../domain/service';
+import { Synchronization } from '../domain';
+import { SynchronizationStatusModel } from '../domain/types';
+import { SynchronizationLoggable } from '../domain/loggable';
+import { SynchronizationErrorLoggableException } from '../domain/loggable-exception';
+import { SynchronizationConfig } from '../config';
+
+@Injectable()
+export class SynchronizationUc {
+	constructor(
+		private readonly configService: ConfigService<SynchronizationConfig, true>,
+		private readonly schulconnexRestClient: SchulconnexRestClient,
+		private readonly synchronizationService: SynchronizationService,
+		private readonly userService: UserService,
+		private readonly logger: Logger
+	) {
+		this.logger.setContext(SynchronizationUc.name);
+	}
+
+	public async updateSystemUsersLastSyncedAt(systemId: string): Promise<void> {
+		this.logger.info(new SynchronizationLoggable('Start synchronization users from systemId', systemId));
+
+		const synchronizationId = await this.synchronizationService.createSynchronization();
+
+		try {
+			const usersToCheck = await this.findUsersToSynchronize(systemId);
+			const chunkSize = this.configService.get<number>('SYNCHRONIZATION_CHUNK');
+			const chunks = this.chunkArray(usersToCheck, chunkSize);
+			const promises = chunks.map((chunk) => this.updateLastSyncedAt(chunk, systemId));
+			const results = await Promise.all(promises);
+			const userSyncCount = results.reduce((acc, curr) => +acc + +curr, 0);
+
+			await this.updateSynchronization(synchronizationId, SynchronizationStatusModel.SUCCESS, userSyncCount);
+			this.logger.info(new SynchronizationLoggable('End synchronization users from systemId', systemId, userSyncCount));
+		} catch (error) {
+			if (error instanceof SynchronizationErrorLoggableException) {
```

Comment:
Why you do this check? I think we should mark the synchronization as `FAILED` in case of any error, not just of a `SynchronizationErrorLoggableException` type. If you did it to be able to call `getLogMessage()` method on it to prepare the error message then you should prepare some default message and replace it with the `getLogMessage()` result only in case `error` is of this `SynchronizationErrorLoggableException`. But please remember that we always want to have the information that the synchronization has failed - even if the error is of a different type.

And for any other `error` type you could prepare a default message like: "unexpected error (first 100 chars): ${ERROR_MESSAGE}". Please remember to truncate the error message to first 100 chars (for an unknown errors). to not put too long (random) messages into the database.

Reply:
done